pkg/d4f/breaker.go

```diff
@@ -0,0 +1,66 @@
+package d4f
+
+import (
+	"context"
+
+	"github.com/kyverno/kyverno/pkg/logging"
+	"github.com/kyverno/kyverno/pkg/metrics"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	sdkmetric "go.opentelemetry.io/otel/metric"
+)
+
+type Breaker interface {
+	Do(context.Context, func(context.Context) error) error
+}
+
+type breaker struct {
+	name  string
+	drops sdkmetric.Int64Counter
+	total sdkmetric.Int64Counter
+	open  func(context.Context) bool
+}
+
+func NewBreaker(name string, open func(context.Context) bool) *breaker {
+	logger := logging.WithName("cricuit-breaker")
+	meter := otel.GetMeterProvider().Meter(metrics.MeterName)
+	drops, err := meter.Int64Counter(
+		"kyverno_admissionreports_drops",
```

Comment:
Just my personal opinion, I would avoid using admissionreports and rename the metric to something like `kyverno_ephermeral_reports_admission_drops`, so that we have rooms to add new metrics for ephemeral reports from background scan, if needed.

Reply:
That's not how metrics work @realshuting ;)

Reply:
On the opposite i would be tempted to call it just `kyverno_breaker_drops`, then all relevant details go into labels

Reply:
Again, I'm talking from a user's perspective. I would literally look for admissionreports just by seeing this metric name.

Reply:
> On the opposite i would be tempted to call it just `kyverno_breaker_drops`, then all relevant details go into labels

This sounds good to me.

Reply:
Done